src/Databases/SQLite/SQLiteUtils.cpp

```diff
@@ -23,35 +23,36 @@ void processSQLiteError(const String & message, bool throw_on_error)
         LOG_ERROR(&Poco::Logger::get("SQLiteEngine"), fmt::runtime(message));
 }
 
-
 String validateSQLiteDatabasePath(const String & path, const String & user_files_path, bool throw_on_error)
 {
-    String canonical_user_files_path = fs::canonical(user_files_path);
-
-    String canonical_path;
-    std::error_code err;
-
     if (fs::path(path).is_relative())
-        canonical_path = fs::canonical(fs::path(user_files_path) / path, err);
-    else
-        canonical_path = fs::canonical(path, err);
+        return fs::absolute(fs::path(user_files_path) / path).lexically_normal();
 
-    if (err)
-        processSQLiteError(fmt::format("SQLite database path '{}' is invalid. Error: {}", path, err.message()), throw_on_error);
+    String absolute_path = fs::absolute(path).lexically_normal();
+    String absolute_user_files_path = fs::absolute(user_files_path).lexically_normal();
 
-    if (!canonical_path.starts_with(canonical_user_files_path))
+    if (!absolute_path.starts_with(absolute_user_files_path))
+    {
         processSQLiteError(fmt::format("SQLite database file path '{}' must be inside 'user_files' directory", path), throw_on_error);
-
-    return canonical_path;
+        return "";
+    }
+    return absolute_path;
 }
 
-
-SQLitePtr openSQLiteDB(const String & database_path, ContextPtr context, bool throw_on_error)
+SQLitePtr openSQLiteDB(const String & path, ContextPtr context, bool throw_on_error)
 {
-    auto validated_path = validateSQLiteDatabasePath(database_path, context->getUserFilesPath(), throw_on_error);
+    auto user_files_path = context->getUserFilesPath();
+    auto database_path = validateSQLiteDatabasePath(path, user_files_path, throw_on_error);
+
+    /// For attach database there is no throw mode.
+    if (database_path.empty())
+        return nullptr;
+
+    if (!fs::exists(database_path))
+        LOG_WARNING(&Poco::Logger::get("SQLite"), "SQLite database path {} does not exist, will create an empty SQLite database", database_path);
```

Comment:
Maybe it's not even a warning
```suggestion
        LOG_DEBUG(&Poco::Logger::get("SQLite"), "SQLite database path {} does not exist, will create an empty SQLite database", database_path);
```